### PR TITLE
the follow-up PR to my bio stuff with some misc stuff which should have been in the release

### PR DESCRIPTION
**Pull request by theidealist101**
getting a bit burned out, I might not do much contributing for a while after this

### CODE REVIEW

**Comment by TheEt1234** [comment]
yikes

**Comment by TheEt1234** [comment]
burnout

File: mods/sbz_bio/habitat.lua [hunk 1]
```diff
@@ -46,28 +46,38 @@ function sbz_api.habitat_tick(start_pos, meta)
     local co2 = 0
     for _, v in ipairs(habitat.co2_sources) do
         local pos, node = unpack(v)
-        co2 = co2+minetest.registered_nodes[node.name].co2_action(pos, node)
+        if stage == PcgRandom(hash(pos)):next(0, 9) then co2 = co2+minetest.registered_nodes[node.name].co2_action(pos, node) end
```
Comment (TheEt1234) [inline mods/sbz_bio/habitat.lua:49]: why use pcgrandom there when math.random exists, just wondering

File: mods/sbz_bio/habitat.lua [hunk 2]
```diff
@@ -46,28 +46,38 @@ function sbz_api.habitat_tick(start_pos, meta)
...
     for _, v in ipairs(habitat.plants) do
         local pos, node, d = unpack(v)
-        if co2-d >= 0 then
-            co2 = co2-d
-            local growth_tick = minetest.registered_nodes[node.name].growth_tick or function(...) end
-            if growth_tick(pos, node) then touched_nodes[hash(pos)] = time end
-        else
-            co2 = 0
-            break
+        if stage == PcgRandom(hash(pos)):next(0, 9) then
```
Comment (TheEt1234) [inline mods/sbz_bio/habitat.lua:65]: same here

File: mods/sbz_bio/habitat.lua (hunk 1 above)
Comment (theidealist101) [reply mods/sbz_bio/habitat.lua:49]: previously it updated everything at once every ten seconds, but I wanted to make it so they'd all update at different times but still always be every ten seconds. I used PcgRandom so I could seed it with the position, and hence each position is updated by the habitat at the same point in each ten second cycle. dunno if that makes sense

Comment (TheEt1234) [reply mods/sbz_bio/habitat.lua:49]: oh ok

**Pull request merged**